### ISSUE TITLE
Regression PR 8826 broke cache.reset()

### ISSUE THREAD

**Issue by tm1000** [apollographql/apollo-client#8851]
**Intended outcome:**
Before #8826 was introduced we used `cache.reset()` to not only reset the cache but also recall all active query watchers. (much like `client.resetStore()`).

However, I noticed that https://github.com/apollographql/apollo-client/pull/8826#discussion_r714166464 seems to break this functionality

This seems to be a regression (Or were we using this wrong) as now the store is reset but no active queries are called again.

**Actual outcome:**
I'd Expect the active query watchers to be called as it worked before

**How to reproduce the issue:**
Call `cache.reset()` before and after #8826

**Comment by tm1000** [comment]
@benjamn if we were using `cache.reset()` wrong before let me know as we can just use `client.resetStore()` instead. Its just we were using cache.reset() inside of the `update` callback in some cases instead of having to manually rebuild caches that changes multiple fields.

**Comment by tm1000** [comment]
@benjamn Thanks for the quick response. I think it fixes the issues but my engineers have told me that `cache.reset()` actually broke between 3.3.21 and 3.4.0 which was rather unexpected (they first told me it was related to this and then they ran through downgrading until they got to 3.3.21 which they said worked)

Looking at the code though it looks like the watchers should refetch even in 3.4.0 and especially after your work in #8826 so the investigation continues but without the work in #8826 it definitely wouldn't have worked at all

**Comment by benjamn** [comment]
Alas, 3.3.21 was the last version before 3.4.0, which was a relatively big minor release. Of all the changes in 3.4.0, https://github.com/apollographql/apollo-client/pull/8107/commits/526c940f3c1298bf0a9a3c3764f9dbe01d4996ee (something I added to #8107, perhaps unwisely) looks the most :suspect: to me.

Please do open a new issue if you find anything more. Now that we have the `Cache.ResetOptions` interface, we can add new options to `cache.reset` relatively easily, to accommodate more use cases simultaneously.

**Comment by tm1000** [comment]
@benjamn Thanks for pointing me in the right direction!
